proto/tendermint/abci/types.proto

```diff
@@ -137,6 +137,7 @@ message RequestPrepareProposal {
   repeated bytes block_data = 1;
   // If an application decides to populate block_data with extra information, they can not exceed this value.
   int64 block_data_size = 2;
+  repeated types.Vote votes = 3;
```

Comment:
is this the previous blocks votes? how come we are passing this up?

Reply:
We need to know about the validator's previous block vote data in order to process related infos inside the PrepareProposal, the main reason is to include the AppDataSelfAuthenticating inside the block header as app_data in case it is needed.

Reply:
could you leave a comment saying this?